src/index.js

```diff
@@ -0,0 +1,155 @@
+const prompt = require('prompt');
+const args = process.argv.splice(process.execArgv.length + 2);
+const axios = require('axios');
+const url = require('url');
+const DEFAULT_PORT = 7000;
+let sessionURL = args[0];
+let port = args[1] || DEFAULT_PORT;
+
+async function start() {
+  console.log('==================== Starting Questinator Connector ==================');
+
+  let loginObject;
+
+  if (!sessionURL) {
+    let result = await getSessionUrlAndPort();
+    sessionURL = result.sessionURL;
+    port = result.port || port;
+  }
+
+  console.log(`Please, run your local server on port ${port}`);
+  const requestInfo = url.parse(sessionURL, true);
+  const WSUrl = `${requestInfo.protocol}//${requestInfo.host}`;
+
+  let connectionTimeout = setTimeout(() => {
+    console.log(`Remote server is unavailable: ${sessionURL}`);
+  }, 15000);
+
+  const socket = require('socket.io-client')(WSUrl);
+
+  socket.on('connect', connectToGame);
+  socket.on('join', onJoin);
+  socket.on('question', onQuestion);
+
+  socket.on('disconnect', function () {
+    console.log(`Disconnected from remote server: ${sessionURL}`);
+    console.log(`Reconnecting...`);
+  });
+
+  async function connectToGame() {
+    clearTimeout(connectionTimeout);
+    console.info(`Connecting to remote server: ${sessionURL}`);
+
+    loginObject = await getLogin();
+
+    socket.emit('join', {
+      login: loginObject.login,
+      sessionURL
+    });
+  }
+
+  function onJoin(data) {
+    if (data.success) {
+      return console.log('You are in the game!'); //tslint:disable-line
+    }
+
+    console.error(`Connection error: ${data.message}`);
+    connectToGame();
+  }
+
+  function onQuestion(data) {
+    console.log(`Question: ${data}`);
+
+    axios
+      .get(`http://localhost:${port}?question=${data}`)
```

Comment:
could we move it to the config or/and add cmd parameter support for it ?

Reply:
fixed